db-scheduler-boot-starter/src/main/java/com/github/kagkarlsson/scheduler/boot/autoconfigure/DbSchedulerAutoConfiguration.java

```diff
@@ -0,0 +1,131 @@
+import org.springframework.boot.actuate.health.HealthIndicatorRegistry;
+import org.springframework.boot.autoconfigure.AutoConfigurationPackage;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.autoconfigure.jdbc.DataSourceAutoConfiguration;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.jdbc.datasource.TransactionAwareDataSourceProxy;
+
+@Configuration
+@EnableConfigurationProperties(DbSchedulerProperties.class)
+@AutoConfigurationPackage
+@AutoConfigureAfter({DataSourceAutoConfiguration.class, HealthIndicatorAutoConfiguration.class})
+@ConditionalOnBean(DataSource.class)
+@ConditionalOnProperty(value = "db-scheduler.enabled", matchIfMissing = true)
+public class DbSchedulerAutoConfiguration {
+    private static final Logger log = LoggerFactory.getLogger(DbSchedulerAutoConfiguration.class);
+    private static Predicate<Task<?>> shouldBeStarted = task -> task instanceof OnStartup;
+
+    private final DbSchedulerProperties config;
+    private final DataSource existingDataSource;
+    private final List<Task<?>> configuredTasks;
+
+    public DbSchedulerAutoConfiguration(DbSchedulerProperties dbSchedulerProperties,
+        DataSource dataSource, List<Task<?>> configuredTasks) {
+        this.config = Objects.requireNonNull(dbSchedulerProperties, "Can't configure DB Scheduler without required configuration");
+        this.existingDataSource = Objects.requireNonNull(dataSource, "An existing javax.sql.DataSource is required");
+        this.configuredTasks = Objects.requireNonNull(configuredTasks, "At least one Task must be configured");
+    }
+
+    /**
+     * Provide an empty customizer if not present in the context.
+     */
+    @ConditionalOnMissingBean
+    @Bean
+    public DbSchedulerCustomizer noopCustomizer() {
+        return new DbSchedulerCustomizer() {
+        };
+    }
+
+    @ConditionalOnBean(DataSource.class)
+    @ConditionalOnMissingBean
+    @Bean(initMethod = "start", destroyMethod = "stop")
+    public Scheduler scheduler(DbSchedulerCustomizer customizer) {
+        log.info("Creating DB Scheduler using tasks from Spring context: {}", configuredTasks);
+
+        if (existingDataSource instanceof TransactionAwareDataSourceProxy) {
+            log.info("Using a transaction aware DataSource");
+        } else {
+            log.info("The configured DataSource is not transaction aware: '{}'. Operations such as schedule, reschedule and cancel will have effect regardless of spanning transactions.", existingDataSource);
```

Comment:
This log statement is a bit confusing. What is the purpose of this?

Reply:
Are there cases where a Spring-user might send in a datasource is not transaction-aware by mistake?

Reply:
@cash1981: I thought that information whether one uses an instance of `TransactionAwareDataSourceProxy` would be useful information, see the [docs](https://github.com/kagkarlsson/db-scheduler#things-to-note--gotchas).

However, I can't seem to get Spring Boot's configuration to provide me with a `TransactionAwareDataSourceProxy` out of the box (using JPA or JDBC starter, `@EnableTransactionManagement` on/off and using a local PostgreSQL db).